src/retasc/jira_client.py

```diff
@@ -0,0 +1,97 @@
+import logging
+
+from atlassian import Jira
+
+logger = logging.getLogger(__name__)
+
+
+class JiraClient:
+    """
+    Jira Client Wrapper
+    """
+
+    def __init__(self, api_url: str, token: str | None = None):
+        self.api_url = api_url
+        self.jira = Jira(
+            url=api_url,
+            token=token,
+        )
+
+    def api_url_issue(self, issue_key: str | None = None) -> str:
+        return f"{self.api_url.rstrip('/')}/rest/api/2/issue/{issue_key or ''}"
+
+    def api_url_create_issue(self) -> str:
+        # The "?updateHistory=false" is the default url for jira.create_issue to connect
+        # with JIRA and the retasc will not use the way of updateHistory=ture at all
+        return f"{self.api_url.rstrip('/')}/rest/api/2/issue?updateHistory=false"
+
+    def api_url_edit_issue(self, issue_key: str, notify_users: bool = True) -> str:
+        return f"{self.api_url.rstrip('/')}/rest/api/2/issue/{issue_key}?notifyUsers={notify_users}"
+
+    def api_url_search_issue(self) -> str:
+        return f"{self.api_url.rstrip('/')}/rest/api/2/search?startAt=0&fields=%2Aall&jql=project+%3D+TEST"
+
```

Comment:
These methods are really confusing. They are part of the interface/API but are only usable in tests.

Remove the methods and in tests it would be best to use specific string literals. E.g.:

```python
requests_mock.post(f"{JIRA_URL}/rest/api/2/issue", json=TEST_RES)
```